posts/2022-03-17-graphql-22004-beta.adoc

```diff
@@ -0,0 +1,327 @@
+* <<java18, Java 18 Support>>
+* <<security_caches, Distributed Security Caches>>
+
+[#graphql]
+== MicroProfile GraphQL 2.0
+
+MicroProfile GraphQL 2.0 incorporates Jakarta EE 9.1 dependencies.  This allows developers to continue using the same functionality provided by MP GraphQL 1.0 but with the updated Jakarta features such as link:{url-prefix}/docs/latest/reference/feature/cdi-3.0.html[CDI 3.0], link:{url-prefix}/docs/latest/reference/feature/restfulWS-3.0.html[restfulWS] 3.0, link:{url-prefix}/docs/latest/reference/feature/jsonb-2.0.html[JSON-B 2.0], etc. Functionally, version 2.0 behaves the same as 1.0 because it uses the same version of the underlying implementation (link:https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL]).  If you'd like to learn more about MP GraphQL, see our previous link:{url-prefix}/blog/2020/06/10/microprofile-graphql-open-liberty.html[blog post about MP GraphQL 1.0] and our link:{url-prefix}/blog/2020/08/28/graphql-apis-open-liberty-20009.html[blog post on how to use the GraphiQL UI and third-party Client APIs].
+
+To enable the MicroProfile GraphQL 2.0 beta feature in your app, add it to your `server.xml`:
+
+[source, xml]
+----
+<featureManager>
+  <feature>mpGraphQL-2.0</feature>
+</featureManager>
+----
+
+[#java18]
+== Java 18 Support
+
+Java 18 is coming soon, and with it the following features and changes:
+
+* link:https://openjdk.java.net/jeps/400[UTF-8 by Default]
+* link:https://openjdk.java.net/jeps/408[Simple Web Server]
+* link:https://openjdk.java.net/jeps/413[Code Snippets in Java API Documentation]
+* link:https://openjdk.java.net/jeps/416[Reimplement Core Reflection with Method Handles]
+* link:https://openjdk.java.net/jeps/417[Vector API (Third Incubator)]
+* link:https://openjdk.java.net/jeps/418[Internet-Address Resolution SPI]
+* link:https://openjdk.java.net/jeps/419[Foreign Function & Memory API (Second Incubator)]
+* link:https://openjdk.java.net/jeps/420[Pattern Matching for switch (Second Preview)]
+* link:https://openjdk.java.net/jeps/421[Deprecate Finalization for Removal]
+
+By trying out Java 18 now, you get more time to review your applications, microservices, and runtime environments and you'll be a step ahead when it becomes generally available.
+
+To give this a try today, link:https://jdk.java.net/18/[download] the early access release of Java 18, <<run, install 22.0.0.4-beta>> version of Open Liberty, edit your Liberty server's link:{url-prefix}/docs/latest/reference/config/server-configuration-overview.html#server-env[server.env file] to point `JAVA_HOME` to your Java 18 installation and start testing!
+
+For more information on Java 18, please visit the Java 18 link:https://jdk.java.net/18/release-notes[release notes page], link:https://download.java.net/java/early_access/jdk18/docs/api/[API Javadoc page] or link:https://jdk.java.net/18/[download page].
+For more information on Open Liberty, please visit our link:{url-prefix}/docs[documentation page].
+
```

Comment:
Now that the POC approved for Java Security to be disabled with Java 18+ (see https://github.com/OpenLiberty/open-liberty/issues/20289) [and this issue came up after I had created the beta blog issue], we probably should mention that here, just so people know that they need to turn it off should they start using Java 18.